pkg/clusteragent/admission/mutate/common/common.go

```diff
@@ -64,31 +64,57 @@ func contains(envs []corev1.EnvVar, name string) bool {
+// InjectEnv injects an env var into a pod template.
+func InjectEnv(pod *corev1.Pod, env corev1.EnvVar) (injected bool) {
+	log.Debugf("Injecting env var '%s' into pod %s", env.Name, PodString(pod))
+
+	for i, _ := range pod.Spec.Containers {
+		injected = injectEnvInContainer(&pod.Spec.Containers[i], env) || injected
+	}
+	for i, _ := range pod.Spec.InitContainers {
+		injected = injectEnvInContainer(&pod.Spec.InitContainers[i], env) || injected
+	}
+
+	return injected
+}
+
+// injectEnvInContainer injects an env var into a container if it doesn't exist.
+func injectEnvInContainer(container *corev1.Container, env corev1.EnvVar) (injected bool) {
+	if contains(container.Env, env.Name) {
+		log.Debugf("Ignoring container '%s': env var '%s' already exist", container.Name, env.Name)
+		return
+	}
+
+	// Prepend rather than append the new variables so that they precede the previous ones in the final list,
+	// allowing them to be referenced in other environment variables downstream.
+	// (See: https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables)
+	container.Env = append([]corev1.EnvVar{env}, container.Env...)
+	return true
+}
+
+// BuildEnvVarFunc is a function that builds a dynamic env var.
+type BuildEnvVarFunc func(container *corev1.Container, init bool) (corev1.EnvVar, error)
+
+// InjectDynamicEnv injects a dynamic env var into a pod template.
+func InjectDynamicEnv(pod *corev1.Pod, fn BuildEnvVarFunc) (injected bool) {
+	log.Debugf("Injecting env var into pod %s", PodString(pod))
+	for i, _ := range pod.Spec.Containers {
+		env, err := fn(&pod.Spec.Containers[i], false)
+		if err != nil {
+			_ = log.Errorf("Error building env var: %v", err)
 			continue
 		}
-		// prepend rather than append so that our new vars precede container vars in the final list, so that they
-		// can be referenced in other env vars downstream.  (see:  Kubernetes dependent environment variables.)
-		pod.Spec.Containers[i].Env = append([]corev1.EnvVar{env}, pod.Spec.Containers[i].Env...)
-		injected = true
-	}
-	for i, ctr := range pod.Spec.InitContainers {
-		if contains(ctr.Env, env.Name) {
-			log.Debugf("Ignoring init container '%s' in pod %s: env var '%s' already exist", ctr.Name, podStr, env.Name)
+		injected = injectEnvInContainer(&pod.Spec.Containers[i], env) || injected
+	}
+	for i, _ := range pod.Spec.InitContainers {
+		env, err := fn(&pod.Spec.Containers[i], true)
+		if err != nil {
+			_ = log.Errorf("Error building env var: %v", err)
 			continue
 		}
-		// prepend rather than append so that our new vars precede container vars in the final list, so that they
-		// can be referenced in other env vars downstream.  (see:  Kubernetes dependent environment variables.)
-		pod.Spec.InitContainers[i].Env = append([]corev1.EnvVar{env}, pod.Spec.InitContainers[i].Env...)
-		injected = true
+		injected = injectEnvInContainer(&pod.Spec.InitContainers[i], env) || injected
```

Comment:
```suggestion
		injected += injectEnvInContainer(&pod.Spec.InitContainers[i], env) || injected
```

Reply:
I've tried but it seems that this does not work.

Reply:
In Go:
```
	a := false
	a += true
```
```
invalid operation: operator + not defined on a (variable of type bool)
```

This seems like a C-specific notation:
```
#include <stdbool.h>
#include <stdio.h>

int main() {
    bool a = false;
    printf("%s\n", a ? "true" : "false");
    a += true;
    printf("%s\n", a ? "true" : "false");
}
```
```
false
true
```